core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -296,6 +296,13 @@ private RelDataType getRelDataType(RelDataTypeFactory typeFactory, Map<String, O
       componentType = toType(typeFactory, component);
       return typeFactory.createArrayType(componentType, -1);
 
+    case MAP:
+      Object key = requireNonNull(map.get("key"), "key");
+      Object value = requireNonNull(map.get("value"), "key");
```

Comment:
```suggestion
      Object key = get(map, "key");
      Object value = get(map, "key");
```

Reply:
Thanks for your view. All comments are addressed.